c++/src/kj/common.h

```diff
@@ -1954,6 +1954,13 @@ class ArrayPtr: public DisallowConstCopyIfNotConst<T> {
     // libc++ std::fill doesn't have memset specialization either.
   }
 
+  inline void copyFrom(kj::ArrayPtr<T> other) {
+    // Copy data from the other array pointer.
+    // Arrays have to be of the same size and memory area MUST NOT overlap.
+    KJ_IREQUIRE(size_ == other.size(), "copy requires arrays of the same size");
```

Comment:
I do also wonder here why these need to be the same size. It should still be safe if `other.size() <= size()`, and just have the method return the actual number of elements copied.

Reply:
I was thinking about it but imo it is much more error-prone this way. It is very simple to size-align slices. But spotting the fact that you don't have enough in the target and are loosing data might be hard. 

I'd rather force clients to check lengths.

Reply:
Fair enough!